app/services/fact_importer.rb

```diff
@@ -39,31 +39,84 @@ def initialize(host, facts = {})
 
   # expect a facts hash
   def import!
+    ensure_fact_names
+
     ActiveRecord::Base.transaction do
       delete_removed_facts
       update_facts
       add_new_facts
     end
 
-    if @error
-      Foreman::Logging.exception("Error during fact import for #{@host.name}", @error)
-      raise ::Foreman::WrappedException.new(@error, N_("Import of facts failed for host %s"), @host.name)
-    end
+    report_error(@error) if @error
     logger.info("Import facts for '#{host}' completed. Added: #{counters[:added]}, Updated: #{counters[:updated]}, Deleted #{counters[:deleted]} facts")
   end
 
+  def report_error(error)
+    Foreman::Logging.exception("Error during fact import for #{@host.name}", error)
+    raise ::Foreman::WrappedException.new(error, N_("Import of facts failed for host %s"), @host.name)
+  end
```

Comment:
This was extracted but never used, can you leave it inline if this was not intentional?

Reply:
It is extracted to reduce ABC metric of `import!`